app/views/layouts/_discussions.html.erb

```diff
@@ -42,22 +42,22 @@
   <% else %>
     <div class="discussions">
       <% @filtered_discussions.each do |discussion| %>
-        <%= link_to item_discussion_path(discussion) do %>
+        <%= link_to exercise_discussion_path(discussion.exercise.id, discussion) do %>
```

Comment:
This one uses the id on purpose. This avoids generating the long link and calling navigable_parents which impacts on performance